### PR TITLE
add sqlite-orm v1.5

### PR DESCRIPTION
**Pull request by hmenn**

### CODE REVIEW

File: meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+SUMMARY = "SQLite ORM light header only library for modern C++"
+HOMEPAGE = "https://github.com/fnc12/sqlite_orm"
+LICENSE = "BSD-3-Clause"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=3b5bf5ee8bb16d8e19359efe11bdc3f2"
+
+inherit cmake
+
+DEPENDS += "sqlite3"
+
+PR = "r1"
```
Comment (kraj) [inline meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb]: Drop setting PR, PR server takes care of it.

File: meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+SRC_URI = "https://github.com/fnc12/sqlite_orm/archive/v1.5.tar.gz"
+SRC_URI[md5sum] = "f50ea3ff480f402bb7d96011643477c6"
+SRC_URI[sha256sum] = "4c77fc203954bfa48d6e9e89b386050599478f8e6c994d37e9a5a823261c52e6"
+
+S = "${WORKDIR}/sqlite_orm-1.5"
+
+EXTRA_OECMAKE += "-DSqliteOrm_BuildTests=OFF"
+
+FILES_${PN}-dev += "${includedir}/* ${libdir}/*"
```
Comment (kraj) [inline meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb]: this seems redundant

Comment (hmenn) [reply meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb]: Project has cmake with install option and tries to install include/sqlite-orm.h and a few cmake file. If we don't add these folder in FILES, bitbake produces an error so I decided to add those files into dev package.

File: meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+inherit cmake
+
+DEPENDS += "sqlite3"
+
+SRC_URI = "https://github.com/fnc12/sqlite_orm/archive/v1.5.tar.gz"
+SRC_URI[md5sum] = "f50ea3ff480f402bb7d96011643477c6"
+SRC_URI[sha256sum] = "4c77fc203954bfa48d6e9e89b386050599478f8e6c994d37e9a5a823261c52e6"
+
+S = "${WORKDIR}/sqlite_orm-1.5"
```
Comment (kraj) [inline meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb]: this should be `${WORKDIR}/sqlite_orm-${PV}`

File: meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+SUMMARY = "SQLite ORM light header only library for modern C++"
+HOMEPAGE = "https://github.com/fnc12/sqlite_orm"
+LICENSE = "BSD-3-Clause"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=3b5bf5ee8bb16d8e19359efe11bdc3f2"
+
+inherit cmake
+
+DEPENDS += "sqlite3"
+
+SRC_URI = "https://github.com/fnc12/sqlite_orm/archive/v1.5.tar.gz"
```
Comment (kraj) [inline meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb]: github archives can be re-generated occasionally and change the checksums, therefore in OE we do not prefer archive, releases are ok. So I would suggest to use 

```
SRCREV = "e8a9e9416f421303f4b8970caab26dadf8bae98b"
SRC_URI = "git://github.com/fnc12/sqlite_orm;protocol=https"
S = "${WORKDIR}/git"
```

File: meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+SRC_URI[md5sum] = "f50ea3ff480f402bb7d96011643477c6"
+SRC_URI[sha256sum] = "4c77fc203954bfa48d6e9e89b386050599478f8e6c994d37e9a5a823261c52e6"
+
+S = "${WORKDIR}/sqlite_orm-${PV}"
```
Comment (kraj) [inline meta-oe/recipes-devtools/sqlite-orm/sqlite-orm_1.5.bb]: Delete all the lines above

**Comment by hmenn** [comment]
Hi @kraj , I changed the code as you requested but I have some issues on `FILES_${PN}-dev += "${includedir}/* ${libdir}/*"`.  I added this line again to fix uninstalled directory issue. By the way, In my yocto image, I don't install sqlite_orm or use as RDEPENDS. I only used as compile time dependency.

```
ERROR: sqlite-orm-1.5-r0 do_package: QA Issue: sqlite-orm: Files/directories were installed but not shipped in any package: 
85   /usr/lib
86   /usr/lib/cmake 
87   /usr/lib/cmake/sqlite_orm 
88   /usr/lib/cmake/sqlite_orm/SqliteOrmConfig.cmake
89   /usr/lib/cmake/sqlite_orm/SqliteOrmTargets.cmake
90   /usr/lib/cmake/sqlite_orm/SqliteOrmConfigVersion.cmake
91 Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
92 sqlite-orm: 6 installed and not shipped files. [installed-vs-shipped]
```

**Comment by kraj** [comment]
> Hi @kraj , I changed the code as you requested but I have some issues on `FILES_${PN}-dev += "${includedir}/* ${libdir}/*"`. I added this line again to fix uninstalled directory issue. By the way, In my yocto image, I don't install sqlite_orm or use as RDEPENDS. I only used as compile time dependency.
> 
> ```
> ERROR: sqlite-orm-1.5-r0 do_package: QA Issue: sqlite-orm: Files/directories were installed but not shipped in any package: 
> 85   /usr/lib
> 86   /usr/lib/cmake 
> 87   /usr/lib/cmake/sqlite_orm 
> 88   /usr/lib/cmake/sqlite_orm/SqliteOrmConfig.cmake
> 89   /usr/lib/cmake/sqlite_orm/SqliteOrmTargets.cmake
> 90   /usr/lib/cmake/sqlite_orm/SqliteOrmConfigVersion.cmake
> 91 Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
> 92 sqlite-orm: 6 installed and not shipped files. [installed-vs-shipped]
> ```

you can add following to recipe

`FILES_${PN}-dev += "${libdir}/cmake/${BPN}"`

**Comment by kraj** [comment]
in master now

**Pull request closed without merging**